src/Php85/Uri/Rfc3986Uri.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Symfony\Polyfill\Php85\Uri;
+
+use Symfony\Polyfill\Php85\Exception\ParsingException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @internal
+ */
+class Rfc3986Uri extends Uri
+{
+    public function __construct(string $uri, ?string $baseUrl = null)
+    {
+        if ('' === trim($uri)) {
+            throw new \ValueError('Argument #1 ($uri) cannot be empty');
```

Comment:
The PHP 8.5 polyfill needs to either depend on the PHP 8.0 polyfill or on PHP 8.0 itself. Otherwise, the `ValueError` class might not be available on PHP 7.

See also the discussions on #499 and #496.

Reply:
Indeed, thank you! I added the polyfill for 8.0, seems the best option to me.

Reply:
Yeah, it's probably the best for the time being. I'm leaning a bit towards making PHP 8.1 a requirement for the PHP 8.5 polyfill package, but we don't have to decide this now.